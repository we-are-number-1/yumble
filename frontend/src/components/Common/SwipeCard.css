/* Swiping page cards */
.SwipeCard{
    outline: none;
    border: 5px solid #000000;
    background-color: white;
    cursor: pointer;
    height: 40vw;
    width: 20vw;
    font-family: Comic Sans MS;
    border-radius: 30px;
    padding: 1.5vw;
    padding-bottom: 0.5vw;
    margin-top: 1vw;
    margin-bottom: 0.5vw;
  }
  
  .CardImage{
    border: 1px solid #000000;
    height: 20vw; 
    width: 20vw; 
    border-radius: 30px;
  }