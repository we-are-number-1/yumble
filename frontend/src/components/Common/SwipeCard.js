import React from 'react';
import './SwipeCard.css';

/**
 * @return {void}
 */
function cardClicked() {
  console.log('Card clicked');
}

/**
*
* @param {*} props
* @return {*}
**/
function SwipeCard(props) {
  const data = props.data;
  const name = data.name;
  const images = data.images;
  const location = data.location;
  const cuisine = data.cuisine;
  const price = data.price;
  const rating = data.rating.toFixed(1);

  return (
    <div className="SwipeCard" onClick = {cardClicked}>
      <div className="CardImage">
        <img className="CardImage" src={images} ></img>
      </div>
      <div>
        <p className='PlaceDetails' id = 'name'> {name} </p>
      </div>
      <p className='PlaceDetails' id= 'location'> {location} </p>
      <p className='PlaceDetails' id = 'price'> {price} </p>
      <p className='PlaceDetails' id = 'cuisine'> {cuisine} </p>
      <p className='PlaceDetails' id = 'rating'> {rating} Rating</p>
    </div>
  );
}
export default SwipeCard;
