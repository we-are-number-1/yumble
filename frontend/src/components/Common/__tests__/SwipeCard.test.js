import React from 'react';
import renderer from 'react-test-renderer';
import SwipeCard from '../SwipeCard';

test('Swiping Card snapshot testing', () => {
  const name = 'Lonestar';
  const images= 'https://c.files.bbci.co.uk/050B/production/_103119210_lazytown2.jpg';
  const location = 'Botany';
  const cuisine = 'European';
  const price = '$$$';
  const rating = 4.0;
  const Data = {name, images, location, cuisine, price, rating};

  const tree = renderer.create(<SwipeCard data = {Data}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
