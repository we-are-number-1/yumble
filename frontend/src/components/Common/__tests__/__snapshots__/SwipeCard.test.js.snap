// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiping Card snapshot testing 1`] = `
<div
  className="SwipeCard"
  onClick={[Function]}
>
  <div
    className="CardImage"
  >
    <img
      className="CardImage"
      src="https://c.files.bbci.co.uk/050B/production/_103119210_lazytown2.jpg"
    />
  </div>
  <div>
    <p
      className="PlaceDetails"
      id="name"
    >
       
      Lonestar
       
    </p>
  </div>
  <p
    className="PlaceDetails"
    id="location"
  >
     
    Botany
     
  </p>
  <p
    className="PlaceDetails"
    id="price"
  >
     
    $$$
     
  </p>
  <p
    className="PlaceDetails"
    id="rating"
  >
     
    4.0
     Rating
  </p>
</div>
`;
